apps/common/controllers/add-another-address-loop.js

```diff
@@ -19,14 +19,28 @@ module.exports = class AddAnotherAddressController extends BaseController {
     const hasAddresses = req.sessionModel.get(`${this.field}Addresses`) ? true : false;
     const items = _.map(addresses, (value, key) => ({
       id: key,
-      address: value.address
+      address: value.address,
+      categories: this.translateCategories(req, value.categories)
     }));
     return Object.assign({}, locals, {
       items,
-      hasAddresses
+      hasAddresses,
+      hasCategories: _.sample(addresses).categories !== undefined
     });
   }
 
+  translateCategories(req, values) {
+    if (!Array.isArray(values)) {
+      values = [values];
+    }
```

Comment:
castArray does exactly the same as these 3 lines of code, however for the sake of duplication it would be better IMO to use the lodash method